include/zephyr/drivers/dma.h

```diff
@@ -256,6 +258,50 @@ struct dma_context {
 /* magic code to identify context content */
 #define DMA_MAGIC 0x47494749
 
+/* Inidicates that a ram region is nocache. */
+/* TODO Obtained from zephyr,memory-region.yaml. This should be defined in
+ * just one place.
+ */
+#define MEM_NOCACHE_ATTR "RAM_NOCACHE"
+
+/**
+ * @brief Container for MMIO SRAM information specified in devicetree
+ *
+ * This type contains pointers to the begin and end of the memory region and
+ * a string that specifies such region's properties.
+ *
+ */
+struct dma_mem_region {
```

Comment:
I still think this would be better represented by a dt_mem_region, its not just dma is it? In device tree memory regions have their own subtree I believe

Reply:
Yes, yesterday night I was still thinking about it and I didn't create the `include/zephyr/devicetree/mem.h` header (mentioned in the previous comment) to avoid having a static inline function in a header file, but I ended up adding it here. As you say, in summary, I will move the SPI function above to its header file so other SPI drivers can use it, and I will move all the code in the `dma.h` header to a new file `include/zephyr/devicetree/mem.h`

Reply:
Since it is declared in `dma.h`, shouldn't it be `dma_dt_mem_region`? As `gpio_dt_spec` does.

Reply:
I'm saying the binding has nothing to do with dma, https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/base/zephyr%2Cmemory-region.yaml it's a general purpose memory region and should be DT related

Reply:
All moved to `include/zephyr/devicetree/mem.h`, as discussed before.